ci/kind/kind-setup.sh

```diff
@@ -352,7 +385,12 @@ while [[ $# -gt 0 ]]
       shift 2
       ;;
     destroy)
-      CLUSTER_NAME="$2"
+      if [[ "$2" == --all && "$3" == --until ]]; then
+         UNTIL=true
+         TIME="$4"
+      else
+        CLUSTER_NAME="$2"
+      fi
```

Comment:
This option parsing is non robust enough, and this is not what we discussed earlier.
Parsing should not break if I swap the order of `--all` and `--until` (for example).
IMO, `--all` and `--until` should be treated as regular options, and parsed as such. At the end of parsing, there should be an error message ("unsupported option") if these options are used incorrectly (e.g., with `create` instead of `destroy`).

Reply:
in existing code (except my changes) if we use these ` --all `and `--until` options with create instead of destroy then it will show unknown option error, but I got your point I will update the code according to error messages.

Reply:
in existing code (except my changes) if we use `--all` and `--until` options with `create` instead of `destroy` then it will show error something like  `Invalid container name (--until-worker), only [a-zA-Z0-9][a-zA-Z0-9_.-] are allowed.` but I got your point I will update the code according to error message.

Reply:
Still not happy with the current solution, so I'm adding a concrete example:
```bash
ACTION=""
CLUSTER_NAME=""
UNTIL_TIME_IN_MINS=""
positional_args=()

function check_action {
    // log error and exit if action is not what's expected
}

while [[ $# -gt 0 ]]
 do
 key="$1"

  case $key in
    create)
      ACTION="create"
      shift
      ;;
    destroy)
      ACTION="destroy"
      shift
      ;;
    --pod-cidr)
      check_action "create"
      POD_CIDR="$2"
      shift 2
      ;;
    --service-cidr)
      check_action "create"
      SERVICE_CIDR="$2"
      shift 2
      ;;
    --ip-family)
      check_action "create"
      IP_FAMILY="$2"
      shift 2
      ;;
    --encap-mode)
      check_action "create"
      ENCAP_MODE="$2"
      shift 2
      ;;
    --no-proxy)
      check_action "create"
      PROXY=false
      shift
      ;;
    --no-kube-proxy)
      check_action "create"
      KUBE_PROXY_MODE="none"
      shift
      ;;
    --prometheus)
      check_action "create"
      PROMETHEUS=true
      shift
      ;;
    --subnets)
      check_action "create"
      SUBNETS="$2"
      shift 2
      ;;
    --images)
      check_action "create"
      IMAGES="$2"
      shift 2
      ;;
    --antrea-cni)
      check_action "create"
      ANTREA_CNI=true
      shift
      ;;
    --num-workers)
      check_action "create"
      NUM_WORKERS="$2"
      shift 2
      ;;
    --k8s-version)
      check_action "create"
      K8S_VERSION="$2"
      shift 2
      ;;
    --all)
      check_action "destroy"
      CLUSTER_NAME="*"
      shift
      ;;
    --until)
      check_action "destroy"
      UNTIL_TIME_IN_MINS="$2"
      shift 2
      ;;
    help)
      print_usage
      exit 0
      ;;
    -*)    # unknown option
      echoerr "Unknown option $1"
      exit 1
      ;;
    *)    # positional arg
      positional_args+=("$1")
      shift
      ;;
 esac
 done

if (( ${#positional_args[@]} > 1 )); then
    echoerr "Too many positional arguments, only expected one (cluster name)"
    exit 1
fi

if (( ${#positional_args[@]} == 1 )) && [[ "$CLUSTER_NAME" == "*" ]]; then
    echoerr "Cannot specify cluster name when using --all"
    exit 1
fi

if (( ${#positional_args[@]} == 1 )); then
    CLUSTER_NAME=${positional_args[0]}
fi

if [[ -z "$CLUSTER_NAME" ]]; then
    echoerr "Missing cluster name"
    exit 1
fi

// at this point, we have done all the validation
// logic goes here for create / destroy
```

Reply:
@antoninbas thanks for the solution, I will update the code.